pkg/ostree/ostree.go

```diff
@@ -72,6 +77,9 @@ type ImageOptions struct {
 	// Indicate if the 'org.osbuild.rhsm.consumer' secret should be added when pulling from the
 	// remote.
 	RHSM bool `json:"rhsm"`
+
+	// TLS verification for container sources.
+	TLSVerify *bool `json:"tls-verify"`
```

Comment:
since there are multiple things here that are "remote" artifacts it might be a little confusing to have this just apply to one of them and still name it `TLSVerify`.

One thing we could do here is break out sets of options that go together into subordinate objects.

i.e.

```
type ContainerImageOptions struct {
    Container string `json:"container"`
    TLSVerify *bool `json:"tls-verify"`
}
...
```

Reply:
There's two things going on here that I don't like and I'm unsure of:
- The `ostree.ImageOptions` and more generally the [`distro.ImageOptions`](https://github.com/osbuild/images/blob/86c3f08097f0d409019f70cc0770fd4bcc564868/pkg/distro/distro.go#L132-L138) that exist separately from things like the blueprint customizations.  These I want to merge into a build config.
- The way that users will be specifying an ostree container for a build when this gets into osbuild-composer.

The way I added the options now sort of reflects things that might happen in composer, but that's just leftover internal structure from the split.

All that said, none of this is considered "stable API" (yet) in images, so I did the quickest thing that felt natural at the time.  You're right though, a completely separate object is nicer.  Getting the "build config" done soon would be even better.

Reply:
I'm OK with whatever.